### PR TITLE
Add holo-dev-server to the dev shell when --holo is passed

### PR DESCRIPTION
**Pull request by r-vdp**
We add a --holo flag to the example subcommand.
When this flag is passed, we add holo-nixpkgs as a flake input and we add holo-dev-server to the dev shell's packages.

This PR builds on #165 that was opened by @alastairong

### CODE REVIEW

**Comment by r-vdp** [comment]
@c12i I pushed a commit to hopefully fix the tests. Weird though that github does not show any log output for the failed runs.

File: run_test.sh
Comment (c12i) [inline run_test.sh:162]: The apostrophe doesn't appear to be correctly escaped, might be what's causing CI to fail

```suggestion
  echo 'holo-dev-server is available in the PATH while it shouldn'\''t'
```

**Review by c12i — CHANGES_REQUESTED** [review]

**Comment by c12i** [comment]
> @c12i I pushed a commit to hopefully fix the tests. Weird though that github does not show any log output for the failed runs.

I noticed this too, not sure why this is the case. However, I ran the tests locally (with above fixes), and the script exits with this error

```sh
holo-dev-server is NOT available in the PATH
```

**Comment by r-vdp** [comment]
> > @c12i I pushed a commit to hopefully fix the tests. Weird though that github does not show any log output for the failed runs.
> 
> I noticed this too, not sure why this is the case. However, I ran the tests locally (with above fixes), and the script exits with this error
> 
> ```shell
> holo-dev-server is NOT available in the PATH
> ```

They were definitely working locally for me. I'll look into it.

**Comment by c12i** [comment]
@r-vdp the tests are still failing after your fix, could you kindly have a look? I would like to support, but I'm unfortunately still new to nix, feel free to follow up with the rest of the team on mattermost for assistance with this if it continues to persist

**Comment by r-vdp** [comment]
Hi @c12i. Well, I know why we get the error, but we are still discussing with some other team members about the most desirable manner of fixing it, since there are some trade-offs to take into account.
So I'm waiting for us to take a final decision before continuing on this PR because there's also the possibility that we'll take another approach.

I will mark the PR as draft for now until we've figured out how we want to continue.

**Comment by r-vdp** [comment]
@c12i we finally got all the required pieces in place at the holo side. Would you be able to give this another look?

**Comment by r-vdp** [comment]
Please review the commits separately, as the first two commits are just the result of running standard formatters to avoid reformatting later on.

File: src/cli.rs
Comment (c12i) [inline src/cli.rs:682]: Thanks for adding this change

**Review by c12i — APPROVED** [review]
LGTM, should probably remove it from draft mode, right @r-vdp?

**Comment by r-vdp** [comment]
@c12i I undrafted and pushed a small change to remove a line from the diff that was not needed anymore.

**Pull request merged**
